### PR TITLE
Windowpainter

### PR DESCRIPTION
**Pull request by thatdanguy23**
## What Does This PR Do
Adds an item that changes the colors of basic windows, reinforced windows (as well as both of their fulltile variants) and windoors. It's based off of the pipe painter code, so you can only choose from the same presets as you would painting pipes, but it's a start.

## Why It's Good For The Game
It's just an extra something to make customizing a bit more fun.

## Images of changes
image
image
image
image

## Changelog
:cl: thatdanguy23
add: added a window painting tool
/:cl:

### CODE REVIEW

File: code/game/objects/items/devices/pipe_painter.dm [hunk 1]
```diff
@@ -33,7 +33,7 @@
 
 
 /obj/item/pipe_painter/attack_self(mob/user as mob)
-	mode = input("Which colour do you want to use?", "Pipe Painter", mode) in modes
+	mode = input("Which colour do you want to use?", name, mode) in modes
```
Comment (thatdanguy23) [inline code/game/objects/items/devices/pipe_painter.dm:36]: Changed to ensure that it said "Window Painter" instead of "Pipe Painter".

**Review by AffectedArc07 — CHANGES_REQUESTED** [review]
Few minor things

**Review by AffectedArc07 — APPROVED** [review]

File: code/game/objects/items/devices/window_painter.dm [hunk 2]
```diff
@@ -0,0 +1,15 @@
+/obj/item/pipe_painter/window_painter
+	name = "window painter"
+	icon_state = "window_painter"
+
+/obj/item/pipe_painter/window_painter/afterattack(atom/A, mob/user as mob)
+	if(!(istype(A,/obj/structure/window/reinforced) || istype(A,/obj/structure/window/basic) || istype(A,/obj/machinery/door/window) || istype(A,/obj/structure/window/full/reinforced) || istype(A,/obj/structure/window/full/basic)) || !in_range(user, A))
```
Comment (SteelSlayer) [inline code/game/objects/items/devices/window_painter.dm:6]: Having so many conditions like this in 1 if statement makes it a bit hard to read, and annoying to add to later. I suggest putting all these types in a list on the window painter, i.e
```dm
var/list/paintable_windows = list(
	/obj/structure/window/reinforced,
	/obj/structure/window/basic
	// etc.
)
```
And then you can do:
```dm
if(!is_type_in_list(A, paintable_windows) || !in_range(user, A))
	return
```

File: code/game/objects/items/devices/window_painter.dm [hunk 3]
```diff
@@ -0,0 +1,22 @@
+/obj/item/pipe_painter/window_painter
+	name = "window painter"
+	icon_state = "window_painter"
+
+var/list/paintable_windows = list(
+		/obj/structure/window/reinforced,
+		/obj/structure/window/basic,
+		/obj/structure/window/full/reinforced,
+		/obj/structure/window/full/basic,
+		/obj/machinery/door/window)
```
Comment (AffectedArc07) [inline code/game/objects/items/devices/window_painter.dm:10]:
```suggestion
	var/list/paintable_windows = list(
			/obj/structure/window/reinforced,
			/obj/structure/window/basic,
			/obj/structure/window/full/reinforced,
			/obj/structure/window/full/basic,
			/obj/machinery/door/window)
```
Needs to be indented to be part of the definition, not a global (unindented) list

**Review by AffectedArc07 — CHANGES_REQUESTED** [review]

**Review by AffectedArc07 — APPROVED** [review]

**Review by SteelSlayer — APPROVED** [review]

**Review by Kyep — APPROVED** [review]
Initially I was not in favor of this, as I felt it would be confusing to have windows that for example looked like plasmaglass but were not.

After thinking about it though, color-coding would be nice and opens up new design options. Not to mention it lets people customize their workspaces. And the available colors (non,red,blue,teal,green,yellow,purple) really don't seem that atrocious.

**Review by AffectedArc07 — APPROVED** [review]

**Pull request merged**